edgekv/lib/edgekv.js

```diff
@@ -232,17 +241,19 @@ export class EdgeKV {
 	 * @param {string} $0.item item key to get from the EdgeKV
 	 * @param {string} [$0.default_value=null] the default value to return if a 404 response is returned from EdgeKV
 	 * @param {number} [$0.timeout=null] the maximum time, between 1 and 1000 milliseconds, to wait for the response
-	 * @returns {string} if the operation was successful, the text response from the EdgeKV or the default_value on 404
+	 * @param {number} [$0.num_retries_on_timeout=null] the number of times to retry a requests when the sub request times out
+	 * @returns {Promise<string>} if the operation was successful, the text response from the EdgeKV or the default_value on 404
 	 * @throws {object} if the operation was not successful,
 	 * 		an object describing the non-200 and non-404 response from the EdgeKV: {failed, status, body}
 	 */
-	async getText({ namespace = this.#namespace, group = this.#group, item, default_value = null, timeout = null } = {}) {
+	async getText({ namespace = this.#namespace, group = this.#group, item, default_value = null, timeout = null, num_retries_on_timeout = null } = {}) {
 		return this.requestHandlerTemplate(
 			() => this.getRequest({ namespace: namespace, group: group, item: item, timeout: timeout }),
 			(response) => response.text(),
 			(response) => this.streamText(response.body),
 			"GET TEXT",
-			default_value
+			default_value,
+			num_retries_on_timeout || this.#num_retries_on_timeout
```

Comment:
Can you confirm the expected behavior if a developer sets the default number of timeouts to a non-zero number in the constructor, but provides `0` as the argument to `num_retries_on_timeout` in `getText()`?  My assumption would be that the code should not retry in this case.

However, it appears the current code would treat the 0 as a signal to use the class-level default instead of overriding the default and preventing retries.

Reply:
Good catch!
Thanks to the magic of JavaScript, it will not work as intended.
I'll fix the logic and update the PR.

Reply:
If I'm not overlooking something, it might be as simple as using the nullish coalescing operator instead.

`num_retries_on_timeout ?? this.#num_retries_on_timeout`

Reply:
JavaScript is new to me (at least since 2007ish) - I've updated with a different solution, it has been tested by QA. I'll update the code to try this for the next update.

Reply:
Going to see if QA is up for another round of testing to get this in now.

Reply:
QA tested and we have updated the code. Ready for approval.